src/features/vault.ts

```diff
@@ -0,0 +1,247 @@
+import * as vscode from 'vscode';
+import * as utilAnsibleCfg from './utils/ansibleCfg';
+import * as tmp from 'tmp';
+import * as cp from 'child_process';
+import * as fs from 'fs';
+
+async function askForVaultId(ansibleCfg: string) {
+  let vaultId = 'default';
+  let identityList: string[] | undefined;
+  if (ansibleCfg === 'ANSIBLE_VAULT_IDENTITY_LIST') {
+    const envVar: string | undefined = process.env.ANSIBLE_VAULT_IDENTITY_LIST;
+    if (envVar !== undefined) {
+      identityList = envVar
+        .split(',')
+        .map((id: string) => id.split('@', 2)[0].trim());
+    } else {
+      identityList = [];
+    }
+  } else {
+    const cfg: utilAnsibleCfg.AnsibleVaultConfig | undefined =
+      utilAnsibleCfg.getValueByCfg(ansibleCfg);
+    if (cfg !== undefined) {
+      identityList = cfg.defaults.vault_identity_list
+        .split(',')
+        .map((id: string) => id.split('@', 2)[0].trim());
+    } else {
+      identityList = [];
+    }
+  }
+  if (identityList.length !== 0) {
+    await vscode.window.showQuickPick(identityList).then((chosen) => {
+      if (chosen) {
+        vaultId = chosen;
+      }
+    });
+    return vaultId;
+  } else {
+    return undefined;
+  }
```

Comment:
Let's use a guard expression here, plus it'd be cleaner to just use await everywhere w/o combining it with then-chains:
```suggestion

  if (!identityList.length) {
    return undefined;
  }

  const chosenVault = await vscode.window.showQuickPick(identityList);
  return chosenVault || vaultId;
```